packages/syft/tests/syft/metadata/metadata_service_test.py

```diff
@@ -0,0 +1,203 @@
+) -> None:
+    # add a mock metadata to the stash
+    mock_metadata = add_mock_metadata(metadata_stash, metadata)
+
+    # get a new metadata according to update_metadata
+    new_metadata = deepcopy(metadata)
+    update_kwargs = update_metadata.to_dict(exclude_none=True).items()
+    for field_name, value in update_kwargs:
+        setattr(new_metadata, field_name, value)
+
+    assert new_metadata != metadata
+    assert new_metadata != mock_metadata
+    assert mock_metadata == metadata
+
+    mock_stash_get_all_output = [mock_metadata, mock_metadata]
+
+    def mock_stash_get_all() -> Ok:
+        return Ok(mock_stash_get_all_output)
+
+    monkeypatch.setattr(metadata_service.stash, "get_all", mock_stash_get_all)
+
+    # update the metadata in the metadata stash using metadata_service
+    response = metadata_service.update(authed_context, update_metadata)
+    updated_metadata = response.ok()[0]
+    not_updated_metadata = response.ok()[1]
+
+    assert response.is_ok() is True
+    assert len(response.ok()) == len(mock_stash_get_all_output)
+    assert updated_metadata == new_metadata  # the first metadata is updated
+    assert not_updated_metadata == metadata  # the second metadata is not updated
+
+    assert True
+
+
+def test_metadataservice_update_stash_get_all_fail(
+    monkeypatch: MonkeyPatch,
+    metadata_service: MetadataService,
+    update_metadata: NodeMetadataUpdate,
+    authed_context: AuthedServiceContext,
+) -> None:
+    # the stash.get_all() function fails
+    mock_error_message = "database failure"
+
+    def mock_stash_get_all_error() -> Err:
+        return Err(mock_error_message)
+
+    monkeypatch.setattr(metadata_service.stash, "get_all", mock_stash_get_all_error)
+    response = metadata_service.update(authed_context, update_metadata)
+
+    assert isinstance(response, SyftError)
+    assert response.message == mock_error_message
+
+
+def test_metadataservice_update_stash_empty(
+    metadata_service: MetadataService,
+    update_metadata: NodeMetadataUpdate,
+    authed_context: AuthedServiceContext,
+) -> None:
+    response = metadata_service.update(authed_context, update_metadata)
+
+    assert isinstance(response, SyftError)
+    assert response.message == "No metadata found"
+
+
+def test_metadataservice_update_fail(
+    monkeypatch: MonkeyPatch,
+    metadata: NodeMetadata,
+    metadata_service: MetadataService,
+    update_metadata: NodeMetadataUpdate,
+    authed_context: AuthedServiceContext,
+) -> None:
+    # the stash has a metadata but we could not update it (the stash.update() function fails)
+
+    mock_stash_get_all_output = [metadata, metadata]
+
+    def mock_stash_get_all() -> Ok:
+        return Ok(mock_stash_get_all_output)
+
+    monkeypatch.setattr(metadata_service.stash, "get_all", mock_stash_get_all)
+
+    mock_update_error_message = "Failed to update obj NodeMetadata"
+
+    def mock_stash_update_error(update_metadata: NodeMetadataUpdate) -> Err:
```

Comment:
The arguments for  `mock_stash_update_error` should be  `metadata: NodeMetadata` instead of NodeMetadataUpdate, as stash update method expect NodeMetadata.